extensions/scarb-execute/src/oracle/jsonrpc.rs

```diff
@@ -0,0 +1,268 @@
+//! Simple JSON-RPC types that suit our needs and nothing more.
+//! It is surprising how complex existing JSON-RPC Rust implementations are,
+//! we don't need any kinds of async, extendibility or network stuff that they provide.
+
+#![allow(dead_code, unused)] // TODO: Remove in next PRs
+
+use anyhow::{Result, bail};
+use serde::de::{Unexpected, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer, de};
+use std::fmt;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(untagged)]
```

Comment:
Should we bail from parsing on additional keys in the messages, since it's out of spec to send them outside params/data/result fields?

Reply:
Can't we just add #[serde(deny_unknown_fields)] to all of the variants? 🤔